### PR TITLE
Orthoganlize

### PR DESCRIPTION
**Pull request by chriselrod**
Maybe some cleanup eventually

### CODE REVIEW

File: src/HermiteNormalForm.jl [hunk 1]
```diff
@@ -46,115 +46,290 @@ end
...
+end
+@inline function zeroSupDiagonal!(A, b, rr, c)
+        M, N = size(A)
+        @inbounds for j = c+1:N
+                Aii = A[rr, c]
+                Aij = A[rr, j]
+                iszero(Aij) && continue
+                if abs(Aii) == 1
+                        @vp for k = 1:M
+                                Ack = A[k, c]
```
Comment (YingboMa) [inline src/HermiteNormalForm.jl]: Oof

File: src/HermiteNormalForm.jl [hunk 2]
```diff
@@ -1,38 +1,38 @@
 module HermiteNormalForm
 
 using LinearAlgebra
-export hnf!, hnf, ishnf
+export hnf!, hnf, ishnf, smith!, smith
```
Comment (YingboMa) [inline src/HermiteNormalForm.jl]: Do we have Smith form computation? :p

Comment (chriselrod) [reply src/HermiteNormalForm.jl]: No

File: src/HermiteNormalForm.jl (hunk 1 above)
Comment (chriselrod) [reply src/HermiteNormalForm.jl]: I had thought I reformatted.

File: src/HermiteNormalForm.jl (hunk 2 above)
Comment (chriselrod) [reply src/HermiteNormalForm.jl]: At one point I was playing around and added an incorrect `smith` function.

**Comment by chriselrod** [comment]
This should definitely be squashed before merging.

**Pull request merged**